### PR TITLE
(4.x.x) Some general code cleanup

### PR DESCRIPTION
**Pull request by adamretter**
Backport of https://github.com/eXist-db/exist/pull/2105

### CODE REVIEW

**Comment by adamretter** [comment]
@dizzzz Ah I missed a small think in the rebase. Should be good now...

File: bin/functions.d/eXist-settings.sh [hunk 1]
```diff
@@ -1,6 +1,8 @@
-# -*-Shell-script-*-
-# Common eXist script functions and settings
-# $Id:eXist-settings.sh 7231 2008-01-14 22:33:35Z wolfgang_m $
+#!/usr/bin/env bash
```
Comment (dizzzz) [inline bin/functions.d/eXist-settings.sh]: A wise (?) man (you?) told me once that is this is a dangerous construct... makes the script vulnerable ?

**Comment by adamretter** [comment]
@dizzzz Rebased as requested

File: bin/functions.d/eXist-settings.sh (hunk 1 above)
Comment (adamretter) [reply bin/functions.d/eXist-settings.sh]: Hmmm. I am not certain about that.

However it seems that these days, `#!/usr/bin/env bash` is recommended over `#!/bin/bash` for portability (not all systems have bash in `/bin` e.g. NixOS).

**Comment by dizzzz** [comment]
:-( the process is really silent for 10+ minutes. De we have a deadlock ?

**Comment by adamretter** [comment]
@dizzzz It was a dangling lock problem.
Somehow whilst merging during the back-port I removed a line that released a lock. I have now reinstated that, rebased and pushed.

All should be good now :-)

**Pull request merged**